pkg/kubectl/cmd/util/factory.go

```diff
@@ -83,6 +83,8 @@ type Factory struct {
 	DefaultNamespace func() (string, bool, error)
 	// Returns the generator for the provided generator name
 	Generator func(name string) (kubectl.Generator, bool)
+	// Check whether the kind of resources could be exposed
+	CanBeExposed func(kind string) error
```

Comment:
kind is probably better

On Wed, Sep 9, 2015 at 3:09 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/util/factory.go
> <https://github.com/kubernetes/kubernetes/pull/13184#discussion_r39012619>
> :
>
> > @@ -83,6 +83,8 @@ type Factory struct {
> >  	DefaultNamespace func() (string, bool, error)
> >  	// Returns the generator for the provided generator name
> >  	Generator func(name string) (kubectl.Generator, bool)
> > +	// Check whether the kind of resources could be exposed
> > +	CanBeExposed func(kind string) error
>
> @smarterclayton <https://github.com/smarterclayton> any strong preference
> between func(kind string) and func(obj runtime.Object)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13184/files#r39012619>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift